homeassistant/components/nanoleaf/light.py

```diff
@@ -210,7 +210,15 @@ def turn_on(self, **kwargs):
                 self._light.brightness = int(brightness / 2.55)
 
         if effect:
-            self._light.effect = effect
+            # Nanoleaf http api returns non-existent effect named "*Solid*" when light set to solid color.
+            # This causes various issues with scening (see https://github.com/home-assistant/core/issues/36359).
+            # Until fixed at the library level, we should ensure the effect exists before sending to nanoleaf API
+            if effect in self._effects_list:
```

Comment:
Please invert this check and raise the error if true. Then we can outdent below.

Reply:
Done